producer.go

```diff
@@ -18,10 +18,11 @@ type Writer interface {
 }
 
 type producer struct {
-	w Writer
+	w           Writer
+	interceptor *ProducerInterceptor
```

Comment:
I think we can support []ProducerInterceptor, in that way client could implement many interceptors if they want

Reply:
I fixed hered. You can check 🙏